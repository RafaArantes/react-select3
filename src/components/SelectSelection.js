import React, { PropTypes } from 'react'

import classNames from 'classnames'
import isFunction from 'lodash/isFunction'

import filterKeyDown from '../utils/filterKeyDown'
import { stopPropagation } from '../utils/events'


const SelectSelection = ({ clearable, formatter, selection = null, placeholder = null, onClearSelection, onKeyDown },
  { cssClassNameSelector }) => {
  let selectionText = null

  if (selection) {
    selectionText = isFunction(formatter) ? formatter(selection) : selection.text
  } else if (placeholder) {
    selectionText = placeholder
  }

  return (
    <span className={ classNames(`${cssClassNameSelector}__selection`, {
      [`${cssClassNameSelector}__selection--placeholder`]: !selection,
      [`${cssClassNameSelector}__selection--clearable`]: clearable,
    })}>
    <span className={`${cssClassNameSelector}__selection-text`}>
      { selectionText }
    </span>
      { clearable && (
        <span className={`${cssClassNameSelector}__clear-selection`}
              role="presentation"
              tabIndex="0"
              onKeyDown={stopPropagation(filterKeyDown([
                { allowedKeys: ['ArrowUp', 'ArrowDown', 'Escape'], func: onKeyDown },
                { allowedKeys: ['Enter', ' '], func: onClearSelection },
              ]))}
              onClick={ stopPropagation(onClearSelection) }>
        &times;
      </span>
      )}
      <span className={`${cssClassNameSelector}__selection-arrow`} role="presentation">
      <i/>
    </span>
  </span>
  )
}

SelectSelection.contextTypes = {
  cssClassNameSelector: PropTypes.string,
}

SelectSelection.propTypes = {
  clearable: PropTypes.bool,
  onClearSelection: PropTypes.func,
  placeholder: PropTypes.string,
  selection: PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }),
  formatter: PropTypes.func,
}

export default SelectSelection
