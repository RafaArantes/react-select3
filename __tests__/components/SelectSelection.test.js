import SelectSelection from '../../src/components/SelectSelection'


describe('SelectSelection', () => {
  it('Should render', () => {
    const component = shallow(<SelectSelection />, selectComponentContext)

    expect(component).toMatchSnapshot()
  })

  it('Should show placeholder when there is no selection provided', () => {
    const placeholder = 'Test placeholder'
    const component = shallow(<SelectSelection placeholder={ placeholder }/>, selectComponentContext)

    expect(component).toMatchSnapshot()
    expect(component.find(`${cssName}__selection-text`).text()).toBe(placeholder)
  })
})
