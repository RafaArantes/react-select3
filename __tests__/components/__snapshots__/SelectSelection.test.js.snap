// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSelection Should render 1`] = `
<span
  className="rs3__selection rs3__selection--placeholder"
>
  <span
    className="rs3__selection-text"
  />
  <span
    className="rs3__selection-arrow"
    role="presentation"
  >
    <i />
  </span>
</span>
`;

exports[`SelectSelection Should show placeholder when there is no selection provided 1`] = `
<span
  className="rs3__selection rs3__selection--placeholder"
>
  <span
    className="rs3__selection-text"
  >
    Test placeholder
  </span>
  <span
    className="rs3__selection-arrow"
    role="presentation"
  >
    <i />
  </span>
</span>
`;
